### PR TITLE
Don't fallback to running as executable unless it's actually executable

### PR DESCRIPTION
**Pull request by lhorie**

### CODE REVIEW

File: rules/execute-command.js [hunk 1]
```diff
@@ -95,13 +95,13 @@ function runCommand(command, args = []) {
...
         .split('${NODE}')
-        .join(`node -r ${join(rootDir, '.pnp.js')}`);
+        .join(`${node} -r ${join(rootDir, '.pnp.js')}`);
     }
     if (command.includes('${ROOT_DIR}')) {
       command = command.split('${ROOT_DIR}').join(rootDir);
     }
     try {
-      exec(command, options);
+      exec(`if [[ -x $(which "${command}") ]]; then "${command}"; fi`, options);
```
Comment (ganemone) [inline rules/execute-command.js:104]: This kind of logic will add additional overhead. Could we instead just say that invalid build commands are not supported?

Comment (lhorie) [reply rules/execute-command.js:104]: I'm fine with removing this side of the branch altogether, but I thought this logic was here to fall back to executing bash commands if a corresponding package.json script was not found. Do we not need this logic anymore?

Comment (ganemone) [reply rules/execute-command.js:104]: We still need this branch, but I don't think we need to add this custom logic. If our build is set up in a way which running `command` fails, the build should just fail.

**Comment by lhorie** [comment]
After some discussion, @ganemone and I decided to leave the current semantics of erroring on non-existent scripts to help surface hidden performance traps

**Pull request closed without merging**